addons/l10n_id/models/res_bank.py

```diff
@@ -76,3 +104,34 @@ def _get_qr_code_generation_params(self, qr_method, amount, currency, debtor_par
                 'value': self._get_qr_vals(qr_method, amount, currency, debtor_partner, free_communication, structured_communication),
             }
         return super()._get_qr_code_generation_params(qr_method, amount, currency, debtor_partner, free_communication, structured_communication)
+
+    def _l10n_id_qris_fetch_status(self, qr_data):
+        """
+        using self and the given data, fetches the status of a specific QR code generated by QRIS
+        Expected values in the qr_data dict are:
+            - invoice_id returned when generating a QR code
+            - the amount present in the qr code
+            - the datetime at which the QR code was generated
+        """
+        return self._l10n_id_make_qris_request(QRIS_FETCH_STATUS_URL, {
+            'do': 'checkStatus',
+            'apikey': self.l10n_id_qris_api_key,
+            'mID': self.l10n_id_qris_mid,
+            'invid': qr_data['qris_invoice_id'],
+            'trxvalue': qr_data['qris_amount'],
+            'trxdate': qr_data['qris_creation_datetime'],
+        })
+
+    @api.model
+    def _l10n_id_make_qris_request(self, url, params):
```

Comment:
Hmm I'd say it's ok but also not ideal indeed.
I'll change it by doing urljoin and instead of taking the whole URL, only the endpoint.
This way it could only ever be used to call request to QRIS rest api.